test/integration/framework/utils.go

```diff
@@ -0,0 +1,358 @@
+// cluster which contains a Kubeconfig that can be used to authenticate against the Shoot cluster. With it,
+// a Kubernetes client for the Shoot cluster will be initialized and attached to
+// the already existing Operation object.
+func (o *GardenerTestOperation) initializeTestShootClients() error {
+	if o.K8sShootClient != nil {
+		return nil
+	}
+
+	k8sShootClient, err := kubernetes.NewClientFromSecret(o.K8sSeedClient, o.Shoot.SeedNamespace, v1beta1.GardenerName, client.Options{})
+	if err != nil {
+		return err
+	}
+
+	o.K8sShootClient = k8sShootClient
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *GardenerTestOperation) generateDeploymentObject(opts *TillerOptions) (*appsv1.Deployment, error) {
+	labels := map[string]string{"name": "tiller"}
+	nodeSelectors := map[string]string{}
+	if len(opts.NodeSelectors) > 0 {
+		err := parseSelectorsIntoMap(opts.NodeSelectors, nodeSelectors)
+		if err != nil {
+			return nil, err
+		}
+	}
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: opts.Namespace,
+			Name:      o.TillerConfigurationOptions.DeploymentName,
+			Labels:    labels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: opts.getReplicas(),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: labels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: labels,
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "tiller",
+							Image:           defaultTillerImage + ":canary",
+							ImagePullPolicy: opts.pullPolicy(),
+							Ports: []corev1.ContainerPort{
+								{ContainerPort: 44134, Name: "tiller"},
+								{ContainerPort: 44135, Name: "http"},
+							},
+							Env: []corev1.EnvVar{
+								{Name: "TILLER_NAMESPACE", Value: opts.Namespace},
+							},
+							LivenessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/liveness",
+										Port: intstr.FromInt(44135),
+									},
+								},
+								InitialDelaySeconds: 1,
+								TimeoutSeconds:      1,
+							},
+							ReadinessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/readiness",
+										Port: intstr.FromInt(44135),
+									},
+								},
+								InitialDelaySeconds: 1,
+								TimeoutSeconds:      1,
+							},
+						},
+					},
+					HostNetwork:  opts.EnableHostNetwork,
+					NodeSelector: nodeSelectors,
+				},
+			},
+		},
+	}
+	return deployment, nil
+}
+
+//TODO: add support to pod-prefixes filtering for logs
+func (o *GardenerTestOperation) fetchPodContainerLogs(podNamespace, podName, containerName string) (*string, error) {
+	logChannel := make(chan string, 1)
+
+	go func() {
+		var (
+			buffer *bytes.Buffer
+			err    error
+		)
+		if len(containerName) == 0 {
```

Comment:
Instead of doing two calls here, just create a single function that does create the `PodLogOptions` for you depending on whether there is a `containerName` or not (or just have one struct and set it if it's non-empty).

Reply:
the whole function is removed, not needed for the test framework